### ISSUE TITLE
Smooth movement

### ISSUE THREAD

**Issue by cdglasz** [kmackenzieii/ai-midterm#14]
So I think that, in order to make movement less choppy (which is a huge problem for time), we would have to do some restructuring. As of right now, the way that the program is structured, the quagent MUST stop at each cell in order to use rays to map. If we try to map as we walk, it screws up the flow of the events, and things stop working. Thoughts?

**Comment by kmackenzieii** [comment]
So for some reason this thread isn't going in my main inbox so I'm not
getting notifications. I saw your smoothing thing though and it works
great. I made a new destination finder that takes into account how
unexplored the whole area is. The quagent does a good job of exploring the
entire space now. It still has a few issues with attempting to walk through
walls that I haven't figured out why yet. Maybe the last focus if we have
time should be on making some better logic for getting unstuck so it
doesn't keep getting stuck over and over.

**Comment by cdglasz** [comment]
So I just looked at the new method for finding destinations and it's brilliant, excellent job. I also tried using the indiscretization path finding for searching (not when unsticking) and it seems to work pretty flawlessly. Also, I *think* I may have sorted out the walking through walls thing (as in every test I've run, it hasn't happened).
However, there's a new problem I've found, which is pretty bad, and connected with the problem I opened this issue about:
![bad](https://cloud.githubusercontent.com/assets/8548538/6766841/dcdab5cc-cfec-11e4-8ae0-166c56ec7405.jpg)
All unexplored cells are being explored, with no new unexplored being added. I'm pretty sure this is because of the fact that we're only using rays when we stop. As long as that's the case, we have to choose between very choppy paths (costing time) and missing unexplored territory
I'm not certain if the null pointer exception is related to this, I haven't done loads of testing on it. This is the only case that I've seen it in, though.

**Comment by cdglasz** [comment]
Also, I agree, we should think of a more elegant solution to getting stuck than turning walking in a random direction
And I think that we should think of a backup destination finding method. Like possibly look for areas of explored space that isn't bordered by walls. In the picture above, it would probably never have found the other room, just because theres no unexplored cells beyond that doorway. We should figure out a way to detect those cases and resolve them

**Comment by kmackenzieii** [comment]
Yeah that would be good as a backup. So like if the unexplored list is
empty then do that. I think the null pointer comes from trying to loop on
elements of an empty list. Just doing a check for that should solve it. I'm
currently at the Hockey East championship in Boston though so I won't get
to it tonight.

**Comment by kmackenzieii** [comment]
Also you probably are but make sure the quagent has max wisdom so it can
use multiple rays.

**Comment by cdglasz** [comment]
I don't think it's that, because that would give an empty stack exception. I'll do some testing tonight
